Modules/Core/include/selxInterfaceAcceptor.hxx

```diff
@@ -20,22 +20,44 @@
 #ifndef InterfaceAcceptor_hxx
 #define InterfaceAcceptor_hxx
 
+#include "selxConnectionInfo.h"
+
 namespace selx
 {
 template< class InterfaceT >
 int
 InterfaceAcceptor< InterfaceT >::Connect( ComponentBase::Pointer providerComponent )
 {
+  // Here the core of the handshake mechanism takes place: One specific 
+  // interface of the Providing Component is taken (by dynamic_cast) and 
+  // passed to the accepting Component.
+  // The function returns the number of successful connects (1 or 0)
+
   std::shared_ptr< InterfaceT > providerInterface = std::dynamic_pointer_cast< InterfaceT >( providerComponent );
   if( !providerInterface )
   {
-    // TODO: Make logger available here and give name of reqeusted interface InterfaceName < InterfaceT >::Get()
-    // this->m_LoggerImpl.Log( ERROR, "Provider does not implement the requested interface.");
+    // casting failed: this Providing Component does not have the specific interface. 
     return 0;
   }
   // connect value interfaces
   this->Accept( providerInterface ); // due to the input argument being uniquely defined in the multiple inheritance tree, all versions of Set() are accessible at component level
-  this->m_isSet = true;
+  // store the interface for access by the user defined component 
+  this->m_AcceptedInterface = providerInterface;
```

Comment:
Could this be moved inside the Accept() function?

Reply:
Preferably not. The Accept() is a virtual function components can override. Storing the acceptedInterfaces is something that should be done in the handshake, whatever the components implement.

Reply:
OK